NRE in HideSoftInputOnTappedChangedManager.UpdateFocusForView

### Description

The problem appears after calling HideSoftInputAsync(CancellationToken.None) and then navigating back


### Steps to Reproduce

I was unable to create a simple scenario to reproduce the problem

### Link to public reproduction project repository

_No response_

### Version with bug

8.0.0-rc.1.9171

### Is this a regression from previous behavior?

Not sure, did not test other versions

### Last version that worked well

Unknown/Other

### Affected platforms

Android

### Affected platform versions

_No response_

### Did you find any workaround?

_No response_

### Relevant log output

```shell
[mono-rt] [ERROR] FATAL UNHANDLED EXCEPTION: System.NullReferenceException: Object reference not set to an instance of an object.
[mono-rt]    at Microsoft.Maui.Controls.HideSoftInputOnTappedChangedManager.UpdateFocusForView(IView _view) in D:\a\_work\1\s\src\Controls\src\Core\ContentPage\HideSoftInputOnTappedChanged\HideSoftInputOnTappedChangedManager.Platform.cs:line 86
[mono-rt]    at Microsoft.Maui.Controls.HideSoftInputOnTappedChangedManager.SetupHideSoftInputOnTapped() in D:\a\_work\1\s\src\Controls\src\Core\ContentPage\HideSoftInputOnTappedChanged\HideSoftInputOnTappedChangedManager.Platform.cs:line 123
[mono-rt]    at Microsoft.Maui.Controls.HideSoftInputOnTappedChangedManager.UpdatePage(ContentPage page) in D:\a\_work\1\s\src\Controls\src\Core\ContentPage\HideSoftInputOnTappedChanged\HideSoftInputOnTappedChangedManager.Platform.cs:line 20
[mono-rt]    at Microsoft.Maui.Controls.ContentPage.UpdateHideSoftInputOnTapped() in D:\a\_work\1\s\src\Controls\src\Core\ContentPage\ContentPage.Mapper.cs:line 24
[mono-rt]    at Microsoft.Maui.Controls.ContentPage.<.ctor>b__8_0(Object _, NavigatedToEventArgs _) in D:\a\_work\1\s\src\Controls\src\Core\ContentPage\ContentPage.cs:line 39
[mono-rt]    at Microsoft.Maui.Controls.Page.SendNavigatedTo(NavigatedToEventArgs args) in D:\a\_work\1\s\src\Controls\src\Core\Page\Page.cs:line 660
[mono-rt]    at Microsoft.Maui.Controls.Shell.SendNavigated(ShellNavigatedEventArgs args) in D:\a\_work\1\s\src\Controls\src\Core\Shell\Shell.cs:line 1155
[mono-rt]    at Microsoft.Maui.Controls.Shell.<.ctor>b__157_0(Object _, ShellNavigatedEventArgs args) in D:\a\_work\1\s\src\Controls\src\Core\Shell\Shell.cs:line 793
[mono-rt]    at Microsoft.Maui.Controls.ShellNavigationManager.<>c__DisplayClass16_0.<HandleNavigated>g__FireNavigatedEvents|3(ShellNavigatedEventArgs a, Shell shell) in D:\a\_work\1\s\src\Controls\src\Core\Shell\ShellNavigationManager.cs:line 274
[mono-rt]    at Microsoft.Maui.Controls.ShellNavigationManager.<>c__DisplayClass16_0.<HandleNavigated>b__2() in D:\a\_work\1\s\src\Controls\src\Core\Shell\ShellNavigationManager.cs:line 264
[mono-rt]    at Microsoft.Maui.Controls.Page.OnAppearing(Action action) in D:\a\_work\1\s\src\Controls\src\Core\Page\Page.cs:line 439
[mono-rt]    at Microsoft.Maui.Controls.BaseShellItem.OnAppearing(Action action) in D:\a\_work\1\s\src\Controls\src\Core\Shell\BaseShellItem.cs:line 179
[mono-rt]    at Microsoft.Maui.Controls.ShellNavigationManager.HandleNavigated(ShellNavigatedEventArgs args) in D:\a\_work\1\s\src\Controls\src\Core\Shell\ShellNavigationManager.cs:line 262
[mono-rt]    at Microsoft.Maui.Controls.ShellNavigationManager.GoToAsync(ShellNavigationParameters shellNavigationParameters, ShellNavigationRequest navigationRequest) in D:\a\_work\1\s\src\Controls\src\Core\Shell\ShellNavigationManager.cs:line 209
[mono-rt]    at Dopnet.YourDive.UI.ViewModels.SpotsViewModel.SelectSpotAsync() in D:\YourDive\Dopnet.YourDive.UI.Logic\YourDive.UI.ViewModels\SpotsViewModel.cs:line 362
[mono-rt]    at Dopnet.Maui.CommandFactory.<>c__DisplayClass7_0.<<CreateCommand>b__0>d.MoveNext() in D:\YourDive\Dopnet.Maui\CommandFactory.cs:line 50
[mono-rt] --- End of stack trace from previous location ---
[mono-rt]    at System.Threading.Tasks.Task.<>c.<ThrowAsync>b__128_0(Object state)
[mono-rt]    at Android.App.SyncContext.<>c__DisplayClass2_0.<Post>b__0() in /Users/runner/work/1/s/xamarin-android/src/Mono.Android/Android.App/SyncContext.cs:line 36
[mono-rt]    at Java.Lang.Thread.RunnableImplementor.Run() in /Users/runner/work/1/s/xamarin-android/src/Mono.Android/Java.Lang/Thread.cs:line 36
[mono-rt]    at Java.Lang.IRunnableInvoker.n_Run(IntPtr jnienv, IntPtr native__this) in /Users/runner/work/1/s/xamarin-android/src/Mono.Android/obj/Release/net8.0/android-34/mcw/Java.Lang.IRunnable.cs:line 84
[mono-rt]    at Android.Runtime.JNINativeWrapper.Wrap_JniMarshal_PP_V(_JniMarshal_PP_V callback, IntPtr jnienv, IntPtr klazz) in /Users/runner/work/1/s/xamarin-android/src/Mono.Android/Android.Runtime/JNINativeWrapper.g.cs:line 26
[HostConnection] HostComposition ext ANDROID_EMU_CHECKSUM_HELPER_v1 ANDROID_EMU_native_sync_v2 ANDROID_EMU_dma_v1 ANDROID_EMU_direct_mem ANDROID_EMU_host_composition_v1 ANDROID_EMU_host_composition_v2 ANDROID_EMU_YUV_Cache ANDROID_EMU_has_shared_slots_host_memory_allocator ANDROID_EMU_sync_buffer_data ANDROID_EMU_read_color_buffer_dma ANDROID_EMU_hwc_multi_configs GL_OES_EGL_image_external_essl3 GL_OES_vertex_array_object GL_KHR_texture_compression_astc_ldr ANDROID_EMU_host_side_tracing ANDROID_EMU_gles_max_version_3_0
```

--- a/src/Controls/src/Core/ContentPage/HideSoftInputOnTappedChanged/HideSoftInputOnTappedChangedManager.Platform.cs
+++ b/src/Controls/src/Core/ContentPage/HideSoftInputOnTappedChanged/HideSoftInputOnTappedChangedManager.Platform.cs
@@ -79,6 +79,14 @@ namespace Microsoft.Maui.Controls
 				return null;
 			}
 
+			if (ve.Window is null)
+			{
+				// This means the xplat IsFocused value has lagged behind navigation events.
+				// This might happen if navigated has fired on the incoming page but the
+				// "LostFocus" event hasn't propagated from the previous one
+				return null;
+			}
+
 			IDisposable? platformToken = SetupHideSoftInputOnTapped(platformView);
 
 #if ANDROID

--- a/src/Controls/tests/DeviceTests/ControlsHandlerTestBase.cs
+++ b/src/Controls/tests/DeviceTests/ControlsHandlerTestBase.cs
@@ -30,6 +30,7 @@ namespace Microsoft.Maui.DeviceTests
 		protected override MauiAppBuilder ConfigureBuilder(MauiAppBuilder mauiAppBuilder)
 		{
 			mauiAppBuilder.Services.AddSingleton<IApplication>((_) => new ApplicationStub());
+			mauiAppBuilder.Services.AddScoped(_ => new HideSoftInputOnTappedChangedManager());
 			return mauiAppBuilder.ConfigureTestBuilder();
 		}
 

--- a/src/Controls/tests/DeviceTests/Elements/Shell/ShellTests.cs
+++ b/src/Controls/tests/DeviceTests/Elements/Shell/ShellTests.cs
@@ -40,6 +40,7 @@ namespace Microsoft.Maui.DeviceTests
 					SetupShellHandlers(handlers);
 					handlers.AddHandler(typeof(NavigationPage), typeof(NavigationViewHandler));
 					handlers.AddHandler(typeof(Button), typeof(ButtonHandler));
+					handlers.AddHandler(typeof(Entry), typeof(EntryHandler));
 					handlers.AddHandler(typeof(Controls.ContentView), typeof(ContentViewHandler));
 					handlers.AddHandler(typeof(ScrollView), typeof(ScrollViewHandler));
 					handlers.AddHandler(typeof(CollectionView), typeof(CollectionViewHandler));
@@ -999,6 +1000,49 @@ namespace Microsoft.Maui.DeviceTests
 
 			await AssertionExtensions.WaitForGC(pageReference);
 			Assert.False(pageReference.IsAlive, "Page should not be alive!");
+		}
+
+		[Fact(DisplayName = "HideSoftInputOnTapped Doesnt Crash If Entry Is Still Focused After Window Is Null")]
+		public async Task HideSoftInputOnTappedDoesntCrashIfEntryIsStillFocusedAfterWindowIsNull()
+		{
+			SetupBuilder();
+
+			Entry entry1;
+			Entry entry2;
+
+			var rootPage = CreatePage(out entry1);
+			var nextPage = CreatePage(out entry2);
+
+			var shell = await CreateShellAsync(shell =>
+			{
+				shell.CurrentItem = rootPage;
+			});
+
+			await CreateHandlerAndAddToWindow<IWindowHandler>(shell, async (handler) =>
+			{
+				entry1.Focus();
+				await entry1.WaitForFocused();
+
+				await rootPage.Navigation.PushAsync(nextPage);
+				entry2.Focus();
+				await entry2.WaitForFocused();
+
+				await shell.GoToAsync("..", false);
+			});
+
+			ContentPage CreatePage(out Entry entry)
+			{
+				entry = new Entry();
+
+				return new ContentPage()
+				{
+					HideSoftInputOnTapped = true,
+					Content = new VerticalStackLayout()
+					{
+						entry
+					}
+				};
+			}
 		}
 
 		[Fact(DisplayName = "Can Reuse Pages")]

